deploy/kubernetes/schema.go

```diff
@@ -0,0 +1,1783 @@
+	),
+	// endregion
+	// region AzureFileVolumeSource
+	schema.NewStructMappedObjectSchema[v1.AzureFileVolumeSource](
+		"AzureFileVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region ConfigMapVolumeSource
+	schema.NewStructMappedObjectSchema[v1.ConfigMapVolumeSource](
+		"ConfigMapVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region VsphereVirtualDiskVolumeSource
+	schema.NewStructMappedObjectSchema[v1.VsphereVirtualDiskVolumeSource](
+		"VsphereVirtualDiskVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region QuobyteVolumeSource
+	schema.NewStructMappedObjectSchema[v1.QuobyteVolumeSource](
+		"QuobyteVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region AzureDiskVolumeSource
+	schema.NewStructMappedObjectSchema[v1.AzureDiskVolumeSource](
+		"AzureDiskVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region PhotonPersistentDiskVolumeSource
+	schema.NewStructMappedObjectSchema[v1.PhotonPersistentDiskVolumeSource](
+		"PhotonPersistentDiskVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region ProjectedVolumeSource
+	schema.NewStructMappedObjectSchema[v1.ProjectedVolumeSource](
+		"ProjectedVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region PortworxVolumeSource
+	schema.NewStructMappedObjectSchema[v1.PortworxVolumeSource](
+		"PortworxVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region ScaleIOVolumeSource
+	schema.NewStructMappedObjectSchema[v1.ScaleIOVolumeSource](
+		"ScaleIOVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region StorageOSVolumeSource
+	schema.NewStructMappedObjectSchema[v1.StorageOSVolumeSource](
+		"StorageOSVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region CSIVolumeSource
+	schema.NewStructMappedObjectSchema[v1.CSIVolumeSource](
+		"CSIVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region EphemeralVolumeSource
+	schema.NewStructMappedObjectSchema[v1.EphemeralVolumeSource](
+		"EphemeralVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+)
+
+func generateVolumeTypeList(except string) []string {
+	l := []string{
+		"hostPath",
+		"emptyDir",
+		"gcePersistentDisk",
+		"awsElasticBlockStore",
+		"secret",
+		"nfs",
+		"iscsi",
+		"glusterfs",
+		"persistentVolumeClaim",
+		"rbd",
+		"flexVolume",
+		"cinder",
+		"cephfs",
+		"flocker",
+		"downwardAPI",
+		"fc",
+		"azureFile",
+		"configMap",
+		"vsphereVolume",
+		"quobyte",
+		"azureDisk",
+		"photonPersistentDisk",
+		"projected",
+		"portworxVolume",
+		"scaleIO",
+		"storageos",
+		"csi",
+		"ephemeral",
+	}
+	var result []string
+	for _, entry := range l {
+		if entry != except {
+			result = append(result, entry)
+		}
+	}
+	return result
+}
+
+var identifier = schema.NewStringSchema(
```

Comment:
Would be good to have some small unit tests to verify those regexpr work as intended - or rather testing the string schema so it includes also min and max. Could you add some?